client/src/Components/Notifications/index.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-unused-vars */
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+const Notifications = ({ username }) => {
+	const [notifications, setNotifications] = useState(
+		localStorage.getItem('notifications')
+			? JSON.parse(localStorage.getItem('notifications'))
+			: []
+	);
+
+	// for opening and EventSource with the server
+	useEffect(() => {
+		if (username) {
+			const eventSource = new EventSource(
+				`http://localhost:3001/api/v1/users/${username}/notifications`,
+				{ withCredentials: true }
+			);
+			eventSource.onmessage = (e) => {
+				addNotification('A comment was added to your post', e.data);
+			};
+		}
+		// removed addNotification as this function reference every rerender
```

Comment:
You're right about the impact of including `addNotification` to the dependencies list.
But to resolve this correctly you need to move this function inside the `useEffect` and update the dependencies accordingly.